st3/lsp_utils/server_npm_resource.py

```diff
@@ -80,13 +83,18 @@ def needs_installation(self) -> bool:
         return True
 
     def install_or_update(self) -> None:
-        shutil.rmtree(self._server_dest, ignore_errors=True)
-        ResourcePath(self._server_src).copytree(self._server_dest, exist_ok=True)
-        dependencies_installed = path.isdir(path.join(self._server_dest, 'node_modules'))
-        if not dependencies_installed:
-            try:
+        try:
+            makedirs(self._server_dest, exist_ok=True)
+            with open(self._installation_marker_file, 'a'):
+                pass
+            if path.isdir(self._server_dest):
+                shutil.rmtree(self._server_dest)
```

Comment:
on line 87 the _server_dest folder is created
just to be deleted on line 91?

What is the reason behind this? :)

Reply:
The `self._installation_marker_file` needs to be created at `_server_dest/..` which is also not created at this point.
So we really care about the parent of `server_dest` to be created and not the `server_dest` itself.

I admit the code is not very clear.